rfcs/0001-messages.md

```diff
@@ -0,0 +1,146 @@
+# FROST messages
+
+Proposes a message layout to exchange information between participants of a FROST setup.
+
+## Motivation
+
+Currently FROST library is complete for 2 round signatures with a dealer/aggregator setup.
+This proposal is only considering that specific features, additions and upgrades will need to be made when DKG is implemented.
+
+Assuming all participants have a FROST library available we need to define message structures in a way that data can be exchanged between participants. The proposal is a collection of data types so each side can do all the actions needed for a real life situation.
+
+## Definitions
+
+- `dealer`
+- `aggergator`
+- `signer`
+- `nonce`
+- `commitment`
+- 
+
+## Guide-level explanation
+
+We propose a message separated in 2 parts, a header and a payload:
+
+```rust
+struct Message {
+    header: Header,
+    payload: Payload,
+}
+```
+
+`Header` will look as follows:
+
+```rust
+struct Header {
+    msg_type: MsgType,
+    version: MsgVersion,
+    sender: Participant,
+    receiver: Participant, 
+}
+```
+
+While `Payload` will be defined as:
+
+```rust
+enum Payload {
+    DealerBroadcast(MsgDealerBroadcast),
+    Commitments(MsgCommitments),
+    SigningPackage(MsgSigningPackage),
+    SignatureShare(MsgSignatureShare),
+    FinalSignature(MsgFinalSignature),
+}
+```
+
+All the messages and new types will be defined in a new file `src/frost/messages.rs`
+
+## Reference-level explanation
+
+Here we explore in detail the header types and all the message payloads. 
+
+### Header
+
+Fields of the header define new types. Proposed implementation for them is as follows:
+
+```rust
+enum MsgType {
```

Comment:
Marking enums with `non_exhaustive` means that you have to add a wildcard (`_`) case when matching, to handle any new message types.

Also you need to define the serialized size of the enum.
(Since it is only 1 byte, you do not need to define byte order.)

```suggestion
#[repr(u8)]
#[non_exhaustive]
enum MsgType {
```

Reply:
Also, what does message type `0` mean?
Does it have any special significance?
Is it always invalid?

Reply:
Message type `0` should be ok and now this is `DealerBroadcast`.